octodns_cloudflare/__init__.py

```diff
@@ -1066,3 +1067,83 @@ def _extra_changes(self, existing, desired, changes):
                 extra_changes.append(Update(existing_record, desired_record))
 
         return extra_changes
+
+class CloudflareException(ProcessorException):
+    pass
+
+class CloudflareProcessor(BaseProcessor):
+    '''
+    Replace Cloudflare proxied values on non Cloudflare providers with the relevant .cdn.cloudflare.net. value.
+
+    Example usage:
+
+    processors:
+        cloudflare:
+            class: octodns_cloudflare.CloudflareProcessor
+
+    ...
+
+    zones:
+        example.com.:
+            ...
+            processors:
+                - cloudflare
+            ...
+    '''
+
+    def __init__(self, name):
+        super().__init__(name)
+
+    def process_source_and_target_zones(self, desired, existing, target):
+
+        # Check if zone is destined for Cloudflare
+        if isinstance(target, CloudflareProvider):
+            # If it is then dont bother with any processing just return now
+            return desired, existing
+
+        for record in desired.records:
+            # Check the record is Cloudflare proxied
+            if record._octodns.get('cloudflare', {}).get('proxied', False):
```

Comment:
Might be cleaner to not the logic here and use a continue rather than having the whole body/other logic nesting multiple levels below it.

```suggestion
            if not record._octodns.get('cloudflare', {}).get('proxied', False):
                # Not interested in this record.
                continue
```

Reply:
Implemented.